src/System.Reflection.Metadata/tests/Metadata/TypeSystem/AssemblyDefinitionTests.cs

```diff
@@ -0,0 +1,148 @@
+
+                Assert.Equal(assembly.Version, assemblyName.Version);
+                Assert.Equal(assembly.Name, assemblyName.Name);
+                Assert.Equal(assembly.ContentType, assemblyName.ContentType);
+
+                ValidateDefinitionAssemblyNameDefaults(assemblyName, hasDefaultCulture: true, hasDefaultHashAlgorithm: true, hasDefaultPublicKey: false, hasDefaultFlags: false);
+                Assert.Equal(assembly.Flags, assemblyName.Flags);
+                Assert.NotNull(assemblyName.GetPublicKeyToken());
+                Assert.NotNull(assemblyName.GetPublicKey());
+
+                ValidateDefinitionAssemblyNameAgainst(assemblyName, reader, assemblyDef);
+
+                Assert.NotEqual(reader.GetString(assemblyRef.Name), assemblyName.Name);
+            }
+        }
+
+        [Fact]
+        public void ValidateAssemblyNameForAssemblyDefinition()
+        {
+            var assemblyItems = new[]
+            {
+                new { Assembly = WinRT.Lib, Name = "Lib", Version = new Version(1, 0, 0, 0), ContentType = AssemblyContentType.WindowsRuntime },
+                new { Assembly = NetModule.AppCS, Name = "AppCS", Version = new Version(1, 2, 3, 4), ContentType = AssemblyContentType.Default },
+                new { Assembly = Namespace.NamespaceTests, Name = "NamespaceTests", Version = new Version(0, 0, 0, 0), ContentType = AssemblyContentType.Default },
+                new { Assembly = PortablePdbs.DocumentsDll, Name = "Documents", Version = new Version(0, 0, 0, 0), ContentType = AssemblyContentType.Default }
+            };
+
+            foreach (var item in assemblyItems)
+            {
+                var reader = MetadataReaderTests.GetMetadataReader(item.Assembly, options: MetadataReaderOptions.ApplyWindowsRuntimeProjections);
+                
+                foreach (var assemblyRefHandle in reader.AssemblyReferences)
+                {
+                    var assemblyDef = reader.GetAssemblyDefinition();
+                    var assemblyRef = reader.GetAssemblyReference(assemblyRefHandle);
+                    var assemblyName = assemblyDef.GetAssemblyName();
+
+                    Assert.Equal(item.Version, assemblyName.Version);
+                    Assert.Equal(item.Name, assemblyName.Name);
+                    Assert.Equal(item.ContentType, assemblyName.ContentType);
+
+                    ValidateDefinitionAssemblyNameDefaults(assemblyName);
+                    ValidateDefinitionAssemblyNameAgainst(assemblyName, reader, assemblyDef);
+
+                    Assert.NotEqual(reader.GetString(assemblyRef.Name), assemblyName.Name);
+                }
+            }
+        }
+
+        private void ValidateDefinitionAssemblyNameDefaults(
+            AssemblyName assemblyName,
+            bool hasDefaultCulture = true, 
+            bool hasDefaultHashAlgorithm = true, 
+            bool hasDefaultPublicKey = true, 
+            bool hasDefaultFlags = true)
```

Comment:
These parameters seem a bit too complicated. 
Do we actually need this function? Why not just compare the actual values against the expected values (default or not)?

Reply:
done